### PR TITLE
consolidate add part methods (e.g. AddEmbeddedPackage) to use one Initpart extension

### PR DESCRIPTION
**Pull request by tomjebo**
This is a new "clean" PR based off main with the #1387 changes reverted via #1404.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -35,6 +35,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - Removed mutable properties on OpenXmlAttribute and marked as `readonly`
 - Removed `OpenXmlPackage.Close` in favor of Dispose (#1373)
 - OpenXmlPart/OpenXmlContainer/OpenXmlPackage and derived types all have internal constructors
+- Consolidate add part methods (e.g. `AddEmbeddedPackagePart`, `AddImagePart`, etc...) to use one InitPart extension (#1391)
```
Comment (twsouthwick) [inline CHANGELOG.md:38]: Please reword this to be what an external user would see

**Review by twsouthwick — CHANGES_REQUESTED** [review]
Let's see if the pattern we discussed offline would work here

File: src/DocumentFormat.OpenXml/Packaging/ThumbnailPartExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace DocumentFormat.OpenXml.Packaging
+{
+    internal static partial class ThumbnailPartExtensions
```
Comment (twsouthwick) [inline src/DocumentFormat.OpenXml/Packaging/ThumbnailPartExtensions.cs:6]: These extensions should be public. Also, can you put them all in a single class for now? Along with the new unit method. We may move them later but it'll be easier to review and make sure we have consistemcy

**Review by twsouthwick — CHANGES_REQUESTED** [review]
Looking good. I'm not a fan of this name either. How about ISupportedRelation<>?

**Comment by twsouthwick** [comment]
Only 10% the size - this will greatly reduce duplication 👍

File: src/DocumentFormat.OpenXml/Packaging/EmbeddedControlPersistenceBinaryDataPartExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace DocumentFormat.OpenXml.Packaging
+{
+    internal static partial class EmbeddedControlPersistenceBinaryDataPartExtensions
+    {
+        /// <summary>
+        /// Adds a EmbeddedControlPersistenceBinaryDataPart to a parent IAddExtensiblePart class.
```
Comment (twsouthwick) [inline src/DocumentFormat.OpenXml/Packaging/EmbeddedControlPersistenceBinaryDataPartExtensions.cs:9]:
```suggestion
        /// Adds a EmbeddedControlPersistenceBinaryDataPart as a relation of the part. 
```

Comment (tomjebo) [reply src/DocumentFormat.OpenXml/Packaging/EmbeddedControlPersistenceBinaryDataPartExtensions.cs]: I made some changes, let me know. I prefer ISupportExtensibleRelationship if that works for you.

Comment (twsouthwick) [reply src/DocumentFormat.OpenXml/Packaging/EmbeddedControlPersistenceBinaryDataPartExtensions.cs]: What's the purpose of extensible in the name?

File: src/DocumentFormat.OpenXml/Packaging/OpenXmlPartExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+using System;
+
+namespace DocumentFormat.OpenXml.Packaging
+{
+    /// <summary>
+    /// Defines extensions for part relationships
+    /// </summary>
+    public static class OpenXmlPartExtensions
+    {
+        internal static OpenXmlPart InitPart(this OpenXmlPartContainer parent, OpenXmlPart childPart, PartTypeInfo partType, string? relId = null)
```
Comment (twsouthwick) [inline src/DocumentFormat.OpenXml/Packaging/OpenXmlPartExtensions.cs:14]: This should be generic for the childpart so it you don't have to cast. Also should be private

File: src/DocumentFormat.OpenXml/Packaging/OpenXmlPartExtensions.cs [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+        {
+            if (parent == null)
+            {
+                throw new ArgumentNullException(message: "Parent part MUST be provided.", paramName: nameof(parent));
```
Comment (twsouthwick) [inline src/DocumentFormat.OpenXml/Packaging/OpenXmlPartExtensions.cs:18]:
```suggestion
                throw new ArgumentNullException(nameof(parent));
```

We generally don't provide a message for this as it's self explanatory.

File: src/DocumentFormat.OpenXml/Packaging/OpenXmlPartExtensions.cs [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+        /// Adds a AlternativeFormatImportPart as a relationship to the parent part
+        /// </summary>
+        /// <param name="parent">The parent part requesting to add. Required.</param>
+        /// <param name="partType">The part type information for the added extensible part. Required.</param>
+        /// <param name="id">The relationship id. Optional, default to null.</param>
+        /// <return>The newly added part</return>
+        public static AlternativeFormatImportPart AddAlternativeFormatImportPart<T>(this T parent, PartTypeInfo partType, string? id = null)
+            where T : OpenXmlPart, ISupportExtensibleRelationship<AlternativeFormatImportPart>
+        {
+            return (AlternativeFormatImportPart)OpenXmlPartExtensions.InitPart(parent, new AlternativeFormatImportPart(), partType, id);
```
Comment (twsouthwick) [inline src/DocumentFormat.OpenXml/Packaging/OpenXmlPartExtensions.cs:45]: => parent.InitPart(new AlternativeFormatImportPart(), partType, id);
```

File: src/DocumentFormat.OpenXml/Packaging/OpenXmlPartExtensions.cs [hunk 7]
```diff
@@ -0,0 +1,214 @@
...
+        }
+
+        /// <summary>
+        /// Adds a CustomPropertyPart as a relationship to the parent part
+        /// </summary>
+        /// <param name="parent">The parent part requesting to add. Required.</param>
```
Comment (twsouthwick) [inline src/DocumentFormat.OpenXml/Packaging/OpenXmlPartExtensions.cs:51]: No need to say that these are required

File: src/DocumentFormat.OpenXml/Packaging/OpenXmlPartExtensions.cs [hunk 8]
```diff
@@ -0,0 +1,214 @@
...
+        /// <param name="id">The relationship id. Optional, default to null.</param>
+        /// <return>The newly added part</return>
+        public static AlternativeFormatImportPart AddAlternativeFormatImportPart<T>(this T parent, PartTypeInfo partType, string? id = null)
+            where T : OpenXmlPart, ISupportExtensibleRelationship<AlternativeFormatImportPart>
+        {
+            return (AlternativeFormatImportPart)OpenXmlPartExtensions.InitPart(parent, new AlternativeFormatImportPart(), partType, id);
+        }
+
+        /// <summary>
+        /// Adds a CustomPropertyPart as a relationship to the parent part
```
Comment (twsouthwick) [inline src/DocumentFormat.OpenXml/Packaging/OpenXmlPartExtensions.cs:49]: Can you use <see cref= syntax?

File: src/DocumentFormat.OpenXml/Packaging/OpenXmlPartExtensions.cs [hunk 9]
```diff
@@ -0,0 +1,214 @@
...
+        /// </summary>
+        public string Extension { get; }
+    }
+
+    /// <summary>
+    /// Defines the interface for tagging a part that can add extensible parts.
+    /// </summary>
+    /// <typeparam name="TPart">Extensible part type that is supported by the implementing class.</typeparam>
+#pragma warning disable CA1040 // Avoid empty interfaces
+    public interface ISupportExtensibleRelationship<TPart>
```
Comment (twsouthwick) [inline src/DocumentFormat.OpenXml/Packaging/OpenXmlPartExtensions.cs:209]: This should be its own file

**Comment by twsouthwick** [comment]
There should also be an overload that handles just a content type string that existed before the change

File: src/DocumentFormat.OpenXml/Packaging/EmbeddedControlPersistenceBinaryDataPartExtensions.cs (hunk 3 above)
Comment (tomjebo) [reply src/DocumentFormat.OpenXml/Packaging/EmbeddedControlPersistenceBinaryDataPartExtensions.cs]: Changed to ISupportRelationship. The other was probably not what I was originally thinking.

File: src/DocumentFormat.OpenXml/Packaging/PresentationDocument.cs [hunk 10]
```diff
@@ -310,29 +309,25 @@ public DigitalSignatureOriginPart AddDigitalSignatureOriginPart()
         }
 
         /// <summary>
-        /// Adds a ThumbnailPart to the PresentationDocument.
+        /// Adds a ThumbnailPart to the WordprocessingDocument
```
Comment (twsouthwick) [inline src/DocumentFormat.OpenXml/Packaging/PresentationDocument.cs:312]: Looks like a copy/paste error.

Why are these not enabled for the interface? You may need to relax the generic constraints to enable this (i.e. these aren't parts)

File: src/DocumentFormat.OpenXml/Packaging/OpenXmlPartExtensions.cs [hunk 11]
```diff
@@ -0,0 +1,424 @@
...
+
+using DocumentFormat.OpenXml.Features;
+using System;
+
+namespace DocumentFormat.OpenXml.Packaging
+{
+    /// <summary>
+    /// Defines extensions for part relationships
+    /// </summary>
+    public static class OpenXmlPartExtensions
```
Comment (twsouthwick) [inline src/DocumentFormat.OpenXml/Packaging/OpenXmlPartExtensions.cs:12]:
```suggestion
    public static class OpenXmlSupportedRelationsionshipExtensions
```

File: src/DocumentFormat.OpenXml/Packaging/OpenXmlPartExtensions.cs [hunk 12]
```diff
@@ -0,0 +1,424 @@
...
+        {
+            if (parent == null)
+            {
+                throw new ArgumentNullException(paramName: nameof(parent));
+            }
+
+            string contentType = partType.ContentType;
+
+            var partExtension = partType.Extension;
+            parent.Features.GetRequired<IPartExtensionFeature>().Register(contentType, partExtension);
```
Comment (twsouthwick) [inline src/DocumentFormat.OpenXml/Packaging/OpenXmlPartExtensions.cs:25]:
```suggestion
            parent.Features.Get<IPartExtensionFeature>()?.Register(contentType, partExtension);
```

If it's not available for some reason, no need to track it

File: src/DocumentFormat.OpenXml/Packaging/OpenXmlPartExtensions.cs [hunk 13]
```diff
@@ -0,0 +1,424 @@
...
+        {
+            AlternativeFormatImportPart childPart = new();
+
+            if (parent == null)
+            {
+                throw new ArgumentNullException(nameof(parent));
+            }
+
+            parent.InitPart(childPart, contentType, id);
+            return childPart;
```
Comment (twsouthwick) [inline src/DocumentFormat.OpenXml/Packaging/OpenXmlPartExtensions.cs:66]:
```suggestion
            if (parent == null)
            {
                throw new ArgumentNullException(nameof(parent));
            }

            return parent.InitPart(new AlternativeFormatImportPart(), contentType, id);
```

And so on for the others

File: src/DocumentFormat.OpenXml/Packaging/OpenXmlPartExtensions.cs [hunk 14]
```diff
@@ -0,0 +1,424 @@
...
+        /// <summary>
+        /// Adds a EmbeddedPackagePart as a relationship to the parent part
+        /// </summary>
+        /// <param name="parent">The parent part requesting to add.</param>
+        /// <param name="partType">The part type information for the added extensible part.</param>
+        /// <param name="id">The relationship id. Optional, default to null.</param>
+        /// <return>The newly added part</return>
+        /// <see cref="EmbeddedPackagePart"/>
+        public static EmbeddedPackagePart AddEmbeddedPackagePart<T>(this T parent, PartTypeInfo partType, string? id = null)
+            where T : OpenXmlPart, ISupportRelationship<EmbeddedPackagePart> => OpenXmlPartExtensions.InitPart<EmbeddedPackagePart>(parent, new EmbeddedPackagePart(), partType, id);
```
Comment (twsouthwick) [inline src/DocumentFormat.OpenXml/Packaging/OpenXmlPartExtensions.cs:243]:
```suggestion
            where T : OpenXmlPart, ISupportRelationship<EmbeddedPackagePart>
            => parent.InitPart(new EmbeddedPackagePart(), partType, id);
```

and so on for the others

**Review by twsouthwick — CHANGES_REQUESTED** [review]
Looking really close - some code clean up asks are left.

I think we're good with the `ISupportRelationship` name for now, but I'm wondering if it should be `ISupportedRelationship<>` as it's a marker for behavior... no need to change, but let's think about it.

File: src/DocumentFormat.OpenXml/Packaging/PresentationDocument.cs (hunk 10 above)
Comment (tomjebo) [reply src/DocumentFormat.OpenXml/Packaging/PresentationDocument.cs:312]: because the are package type parts which is why they weren't generating add*part methods previously as well. I decided to leave them as is because my first attempt to pull them (spreadsheetdocument, wordprocessingdocument and presentationdocument) into the new generated code met issues. I can make another attempt with the partwriter.

Comment (tomjebo) [reply src/DocumentFormat.OpenXml/Packaging/PresentationDocument.cs:312]: ah, yeah, the methods were manually added and that's what I was referring to and why I figured there was a reason they didn't try to generate for the package parts. But I see what you mean, I'll add the interface derivation in the handwritten part cs. thanks.

Comment (tomjebo) [reply src/DocumentFormat.OpenXml/Packaging/PresentationDocument.cs:312]: brilliant! ;)

File: src/DocumentFormat.OpenXml/Packaging/OpenXmlPartExtensions.cs (hunk 14 above)
Comment (tomjebo) [reply src/DocumentFormat.OpenXml/Packaging/OpenXmlPartExtensions.cs:243]: that's the way I had it initially. ok, I'll change them all back. :/

File: src/DocumentFormat.OpenXml/Packaging/OpenXmlPartExtensions.cs (hunk 13 above)
Comment (tomjebo) [reply src/DocumentFormat.OpenXml/Packaging/OpenXmlPartExtensions.cs:66]: parent doesn't return a part, that's why I added the childPart local.

File: src/DocumentFormat.OpenXml/Packaging/OpenXmlSupportedRelationshipExtensions.cs [hunk 15]
```diff
@@ -0,0 +1,435 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using DocumentFormat.OpenXml.Features;
+using System;
+
+namespace DocumentFormat.OpenXml.Packaging
```
Comment (twsouthwick) [inline src/DocumentFormat.OpenXml/Packaging/OpenXmlSupportedRelationshipExtensions.cs:7]: nit: can you do a namespace scoped file? i.e. add a semicolon to the end here and remove the tab/braces

File: src/DocumentFormat.OpenXml/Packaging/OpenXmlSupportedRelationshipExtensions.cs [hunk 16]
```diff
@@ -0,0 +1,435 @@
...
+        /// </summary>
+        /// <param name="parent">The parent part requesting to add.</param>
+        /// <param name="contentType">The content type information for the added extensible part.</param>
+        /// <param name="id">The relationship id. Optional, default to null.</param>
+        /// <return>The newly added part</return>
+        /// <see cref="CustomPropertyPart"/>
+        public static CustomPropertyPart AddCustomPropertyPart<T>(this T parent, string contentType, string? id = null)
+            where T : OpenXmlPartContainer, ISupportedRelationship<CustomPropertyPart>
+        {
+            CustomPropertyPart childPart = new();
```
Comment (twsouthwick) [inline src/DocumentFormat.OpenXml/Packaging/OpenXmlSupportedRelationshipExtensions.cs:93]: please inline this

File: src/DocumentFormat.OpenXml/Packaging/OpenXmlSupportedRelationshipExtensions.cs [hunk 17]
```diff
@@ -0,0 +1,435 @@
...
+        /// </summary>
+        /// <param name="parent">The parent part requesting to add.</param>
+        /// <param name="contentType">The content type information for the added extensible part.</param>
+        /// <param name="id">The relationship id. Optional, default to null.</param>
+        /// <return>The newly added part</return>
+        /// <see cref="AlternativeFormatImportPart"/>
+        public static AlternativeFormatImportPart AddAlternativeFormatImportPart<T>(this T parent, string contentType, string? id = null)
+            where T : OpenXmlPartContainer, ISupportedRelationship<AlternativeFormatImportPart>
+        {
+            AlternativeFormatImportPart childPart = new();
```
Comment (twsouthwick) [inline src/DocumentFormat.OpenXml/Packaging/OpenXmlSupportedRelationshipExtensions.cs:59]: please inline this

File: src/DocumentFormat.OpenXml/Packaging/OpenXmlSupportedRelationshipExtensions.cs [hunk 18]
```diff
@@ -0,0 +1,435 @@
...
+        {
+            CustomXmlPart childPart = new();
+
+            if (parent == null)
+            {
+                throw new ArgumentNullException(nameof(parent));
+            }
+
+            parent.InitPart(childPart, contentType, id);
+            return childPart;
```
Comment (twsouthwick) [inline src/DocumentFormat.OpenXml/Packaging/OpenXmlSupportedRelationshipExtensions.cs:135]:
```suggestion
            if (parent == null)
            {
                throw new ArgumentNullException(nameof(parent));
            }

            return parent.InitPart(childPart, contentType, id);
```

File: src/DocumentFormat.OpenXml/Packaging/OpenXmlSupportedRelationshipExtensions.cs [hunk 19]
```diff
@@ -0,0 +1,435 @@
...
+        }
+    }
+
+    /// <summary>
+    /// Defines information used in creating a new part.
+    /// </summary>
+    public readonly record struct PartTypeInfo
```
Comment (twsouthwick) [inline src/DocumentFormat.OpenXml/Packaging/OpenXmlSupportedRelationshipExtensions.cs:414]: this should be its own file

File: src/DocumentFormat.OpenXml/Packaging/OpenXmlSupportedRelationshipExtensions.cs (hunk 18 above)
Comment (tomjebo) [reply src/DocumentFormat.OpenXml/Packaging/OpenXmlSupportedRelationshipExtensions.cs:135]: parent's InitPart returns void.

File: src/DocumentFormat.OpenXml/Packaging/OpenXmlPartExtensions.cs (hunk 13 above)
Comment (twsouthwick) [reply src/DocumentFormat.OpenXml/Packaging/OpenXmlPartExtensions.cs]: Just saw this. You can add a helper method here that does that in this class. It makes a lot more verbose

File: src/DocumentFormat.OpenXml/Packaging/OpenXmlSupportedRelationshipExtensions.cs (hunk 18 above)
Comment (tomjebo) [reply src/DocumentFormat.OpenXml/Packaging/OpenXmlSupportedRelationshipExtensions.cs:135]: nm, used generic helper.

**Comment by twsouthwick** [comment]
build is failing....

File: src/DocumentFormat.OpenXml/Packaging/OpenXmlSupportedRelationshipExtensions.cs [hunk 20]
```diff
@@ -0,0 +1,312 @@
...
+    /// <param name="partType">The part type information for the added extensible part.</param>
+    /// <param name="id">The relationship id. Optional, default to null.</param>
+    /// <return>The newly added part</return>
+    /// <see cref="EmbeddedPackagePart"/>
+    public static EmbeddedPackagePart AddEmbeddedPackagePart<T>(this T parent, PartTypeInfo partType, string? id = null)
+        where T : OpenXmlPartContainer, ISupportedRelationship<EmbeddedPackagePart>
+        => InitPart(parent, new EmbeddedPackagePart(), partType, id);
+
+    /// <summary>
+    /// Adds a EmbeddedPackagePart as a relationship to the parent part
```
Comment (twsouthwick) [inline src/DocumentFormat.OpenXml/Packaging/OpenXmlSupportedRelationshipExtensions.cs:206]:
```suggestion
    /// Adds a <see cref="EmbeddedPackagePart" /> as a relationship to the parent part
```

and elsewhere in the doc

**Comment by tomjebo** [comment]
> build is failing....

yeah, no idea what that's about. Any ideas?

**Comment by tomjebo** [comment]
> > build is failing....
> 
> yeah, no idea what that's about. Any ideas?

The errors are all this: 

##[error]Error: The process 'C:\hostedtoolcache\windows\dotnet\dotnet.exe' failed with exit code 1
##[warning].NET 5 has some compatibility issues with older Nuget versions(<=5.7), so if you are using an older Nuget version(and not dotnet cli) to restore, then the dotnet cli commands (e.g. dotnet build) which rely on such restored packages might fail. To mitigate such error, you can either: (1) - Use dotnet cli to restore, (2) - Use Nuget version 5.8 to restore, (3) - Use global.json using an older sdk version(<=3) to build
Info: Azure Pipelines hosted agents have been updated and now contain .Net 5.x SDK/Runtime along with the older .Net Core version which are currently lts. Unless you have locked down a SDK version for your project(s), 5.x SDK might be picked up which might have breaking behavior as compared to previous versions. You can learn more about the breaking changes here: https://docs.microsoft.com/en-us/dotnet/core/tools/ and https://docs.microsoft.com/en-us/dotnet/core/compatibility/ . To learn about more such changes and troubleshoot, refer here: https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/build/dotnet-core-cli?view=azure-devops#troubleshooting
##[error]Dotnet command failed with non-zero exit code on the following projects : [ '' ]

And on different projects some of which I don't think I've touched.

**Comment by twsouthwick** [comment]
You're getting a stackoverflow, so test reporting fails:

```
   at System.Reflection.MethodInvoker.Invoke(System.Object, IntPtr*, System.Reflection.BindingFlags)
   at System.Reflection.RuntimeMethodInfo.Invoke(System.Object, System.Reflection.BindingFlags, System.Reflection.Binder, System.Object[], System.Globalization.CultureInfo)
   at Xunit.Sdk.TestInvoker`1+<>c__DisplayClass48_0+<<InvokeTestMethodAsync>b__1>d[[System.__Canon, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].MoveNext()
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[[Xunit.Sdk.TestInvoker`1+<>c__DisplayClass48_0+<<InvokeTestMethodAsync>b__1>d[[System.__Canon, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]], xunit.execution.dotnet, Version=2.4.2.0, Culture=neutral, PublicKeyToken=8d05b1bb7a6fdb6c]](<<InvokeTestMethodAsync>b__1>d<System.__Canon> ByRef)
   at Xunit.Sdk.TestInvoker`1+<>c__DisplayClass48_0[[System.__Canon, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].<InvokeTestMethodAsync>b__1()
   at Xunit.Sdk.ExecutionTimer+<AggregateAsync>d__4.MoveNext()
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[[Xunit.Sdk.ExecutionTimer+<AggregateAsync>d__4, xunit.execution.dotnet, Version=2.4.2.0, Culture=neutral, PublicKeyToken=8d05b1bb7a6fdb6c]](<AggregateAsync>d__4 ByRef)
   at Xunit.Sdk.ExecutionTimer.AggregateAsync(System.Func`1<System.Threading.Tasks.Task>)
   at Xunit.Sdk.ExceptionAggregator+<RunAsync>d__9.MoveNext()
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[[Xunit.Sdk.ExceptionAggregator+<RunAsync>d__9, xunit.core, Version=2.4.2.0, Culture=neutral, PublicKeyToken=8d05b1bb7a6fdb6c]](<RunAsync>d__9 ByRef)
   at Xunit.Sdk.ExceptionAggregator.RunAsync(System.Func`1<System.Threading.Tasks.Task>)
   at Xunit.Sdk.TestInvoker`1+<InvokeTestMethodAsync>d__48[[System.__Canon, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].MoveNext()
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[[Xunit.Sdk.TestInvoker`1+<InvokeTestMethodAsync>d__48[[System.__Canon, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]], xunit.execution.dotnet, Version=2.4.2.0, Culture=neutral, PublicKeyToken=8d05b1bb7a6fdb6c]](<InvokeTestMethodAsync>d__48<System.__Canon> ByRef)
   at Xunit.Sdk.TestInvoker`1[[System.__Canon, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].InvokeTestMethodAsync(System.Object)
   at Xunit.Sdk.TestInvoker`1+<<RunAsync>b__47_0>d[[System.__Canon, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].MoveNext()
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[[Xunit.Sdk.TestInvoker`1+<<RunAsync>b__47_0>d[[System.__Canon, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]], xunit.execution.dotnet, Version=2.4.2.0, Culture=neutral, PublicKeyToken=8d05b1bb7a6fdb6c]](<<RunAsync>b__47_0>d<System.__Canon> ByRef)
   at Xunit.Sdk.TestInvoker`1[[System.__Canon, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].<RunAsync>b__47_0()
   at Xunit.Sdk.ExceptionAggregator+<RunAsync>d__10`1[[System.Decimal, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].MoveNext()
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[[Xunit.Sdk.ExceptionAggregator+<RunAsync>d__10`1[[System.Decimal, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]], xunit.core, Version=2.4.2.0, Culture=neutral, PublicKeyToken=8d05b1bb7a6fdb6c]](<RunAsync>d__10`1<System.Decimal> ByRef)
   at Xunit.Sdk.ExceptionAggregator.RunAsync[[System.Decimal, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]](System.Func`1<System.Threading.Tasks.Task`1<System.Decimal>>)
   at Xunit.Sdk.XunitTestRunner+<InvokeTestAsync>d__4.MoveNext()
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[[Xunit.Sdk.XunitTestRunner+<InvokeTestAsync>d__4, xunit.execution.dotnet, Version=2.4.2.0, Culture=neutral, PublicKeyToken=8d05b1bb7a6fdb6c]](<InvokeTestAsync>d__4 ByRef)
   at Xunit.Sdk.XunitTestRunner.InvokeTestAsync(Xunit.Sdk.ExceptionAggregator)
   at Xunit.Sdk.ExceptionAggregator+<RunAsync>d__10`1[[System.__Canon, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].MoveNext()
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[[Xunit.Sdk.ExceptionAggregator+<RunAsync>d__10`1[[System.__Canon, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]], xunit.core, Version=2.4.2.0, Culture=neutral, PublicKeyToken=8d05b1bb7a6fdb6c]](<RunAsync>d__10`1<System.__Canon> ByRef)
   at Xunit.Sdk.ExceptionAggregator.RunAsync[[System.__Canon, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]](System.Func`1<System.Threading.Tasks.Task`1<System.__Canon>>)
   at Xunit.Sdk.TestRunner`1+<RunAsync>d__43[[System.__Canon, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].MoveNext()
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[[Xunit.Sdk.TestRunner`1+<RunAsync>d__43[[System.__Canon, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]], xunit.execution.dotnet, Version=2.4.2.0, Culture=neutral, PublicKeyToken=8d05b1bb7a6fdb6c]](<RunAsync>d__43<System.__Canon> ByRef)
   at Xunit.Sdk.TestRunner`1[[System.__Canon, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].RunAsync()
   at Xunit.Sdk.TestCaseRunner`1+<RunAsync>d__19[[System.__Canon, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].MoveNext()
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[[Xunit.Sdk.TestCaseRunner`1+<RunAsync>d__19[[System.__Canon, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]], xunit.execution.dotnet, Version=2.4.2.0, Culture=neutral, PublicKeyToken=8d05b1bb7a6fdb6c]](<RunAsync>d__19<System.__Canon> ByRef)
   at Xunit.Sdk.TestCaseRunner`1[[System.__Canon, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].RunAsync()
   at Xunit.Sdk.TestMethodRunner`1+<RunTestCasesAsync>d__32[[System.__Canon, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].MoveNext()
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[[Xunit.Sdk.TestMethodRunner`1+<RunTestCasesAsync>d__32[[System.__Canon, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]], xunit.execution.dotnet, Version=2.4.2.0, Culture=neutral, PublicKeyToken=8d05b1bb7a6fdb6c]](<RunTestCasesAsync>d__32<System.__Canon> ByRef)
   at Xunit.Sdk.TestMethodRunner`1[[System.__Canon, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].RunTestCasesAsync()
   at Xunit.Sdk.TestMethodRunner`1+<RunAsync>d__31[[System.__Canon, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].MoveNext()
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[[Xunit.Sdk.TestMethodRunner`1+<RunAsync>d__31[[System.__Canon, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]], xunit.execution.dotnet, Version=2.4.2.0, Culture=neutral, PublicKeyToken=8d05b1bb7a6fdb6c]](<RunAsync>d__31<System.__Canon> ByRef)
   at Xunit.Sdk.TestMethodRunner`1[[System.__Canon, System.Private.CoreLib, Version=7.0.0.0, Culture=neutral, PublicKe
Results File: /home/vsts/work/_temp/_fv-az708-261_2023-04-20_20_02_49[1].trx
```

**Comment by twsouthwick** [comment]
That didn't copy well.

Check this permalink: https://office.visualstudio.com/OC/_build/results?buildId=19674033&view=logs&j=79d20d22-154c-59db-67c6-913688a42376&t=ffaec100-2b48-55c6-efba-27bfce93c0c5&l=86

**Comment by tomjebo** [comment]
> That didn't copy well.
> 
> Check this permalink: https://office.visualstudio.com/OC/_build/results?buildId=19674033&view=logs&j=79d20d22-154c-59db-67c6-913688a42376&t=ffaec100-2b48-55c6-efba-27bfce93c0c5&l=86

Ok, I see it. That test passes locally using .net 7.

File: src/DocumentFormat.OpenXml/Packaging/PresentationDocument.cs [hunk 21]
```diff
@@ -309,58 +308,6 @@ public DigitalSignatureOriginPart AddDigitalSignatureOriginPart()
...
-        /// </summary>
-        /// <typeparam name="T">The class of the part.</typeparam>
-        /// <param name="contentType">The content type of the part. Must match the defined content type if the part is fixed content type.</param>
-        /// <param name="id">The relationship id. The id will be automatically generated if this param is null.</param>
-        /// <returns>The added part.</returns>
-        /// <exception cref="OpenXmlPackageException">When the part is not allowed to be referenced by this part.</exception>
-        /// <exception cref="ArgumentOutOfRangeException">When the part is fixed content type and the passed in contentType does not match the defined content type.</exception>
-        /// <exception cref="ArgumentNullException">Thrown when "contentType" is null reference.</exception>
-        /// <remarks>Mainly used for adding not-fixed content type part - ImagePart, etc.</remarks>
-        public override T AddNewPart<T>(string contentType, string id)
```
Comment (twsouthwick) [inline src/DocumentFormat.OpenXml/Packaging/PresentationDocument.cs:349]: why'd you remove this? it does a check that ensures only the main part is the correct content type

Comment (twsouthwick) [reply src/DocumentFormat.OpenXml/Packaging/PresentationDocument.cs:349]: Doesn't look like you deleted the equivalent in the others, so this is probably a mistake

**Review by twsouthwick — APPROVED** [review]
LGTM!
